dask/bag/tests/test_bag.py

```diff
@@ -370,6 +370,22 @@ def test_foldby():
     assert set(c) == set(reduceby(iseven, lambda acc, x: acc + x, L, 0).items())
 
 
+def test_foldby_tree_reduction():
+    for n in [1, 7, 32]:
+        b = db.from_sequence(range(100), npartitions=n)
+        c = b.foldby(iseven, add)
+        r = c.compute()
+        for m in [False, None, 2, 3]:
+            d = b.foldby(iseven, add, split_every=m)
+            e = b.foldby(iseven, add, 0, split_every=m)
+            f = b.foldby(iseven, add, 0, add, split_every=m)
+            g = b.foldby(iseven, add, 0, add, 0, split_every=m)
+            assert r == d.compute()
+            assert r == e.compute()
+            assert r == f.compute()
+            assert r == g.compute()
```

Comment:
Cool.  This provides way more confidence.  However I also notice that this takes a few seconds to run, which is significant given that the entire dask.bag test suite currently runs in about 12 seconds.  

We can accelerate this at least a little bit by using the single-threaded scheduler rather than the multiprocessing scheduler (the default for bag).

You can change this by wrapping your code in the following line:

```python
with dask.set_options(get=dask.local.get_sync):
```

If you have any other thoughts on how to reduce testing time while still maintaining test coverage that would also be welcome.

Reply:
I used dask to compute the results of the test. In my tests it ran a lot faster. What do you think?